public/pages/CreateMonitor/containers/DefineMonitor/DefineMonitor.js

```diff
@@ -220,35 +259,51 @@ class DefineMonitor extends Component {
     const { dataTypes, response, performanceResponse } = this.state;
     const { index, searchType, timeField } = values;
     const isGraph = searchType === SEARCH_TYPE.GRAPH;
+    const isHTTP = searchType === SEARCH_TYPE.HTTP;
+    // Definition of when the "run" button should be disabled for HTTP type.
+    const disableHTTP =
+      (isHTTP && (values.http.urlType === URL_TYPE.FULL_URL && !values.http.url)) ||
+      (values.http.urlType === URL_TYPE.ATTRIBUTE_URL && !values.http.host);
     let invalidJSON = false;
     try {
       JSON.parse(values.query);
     } catch (e) {
       invalidJSON = true;
     }
-    const runIsDisabled = invalidJSON || !values.index.length;
+
+    const runIsDisabled = invalidJSON || (!isHTTP && !values.index.length) || disableHTTP;
```

Comment:
If I'm on extraction query and enter invalid JSON which disables the run button, and then switch to http will the invalidJSON clear or stay as invalid and keeps the run button disabled?

HTTP -> Http

Reply:
Yes it will. Thanks for checking this, I will change this line to `const runIsDisabled = (!isHttp && invalidJSON || !values.index.length) || disableHttp;` along with the HTTP-> Http change on variable names.

Reply:
That doesn't seem like it'll do what you want.
If it is in fact Http then it will evaluate as
(!true && invalidJSON || !values.index.length) => (false && boolean || boolean)
Regardless of that isHttp check it will always evaluate that last boolean.
Perhaps change it to something like:
`const runIsDisabled = isHttp ? disableHttp : (invalidJSON || !values.index.length)`

Reply:
Got it, changed to `const runIsDisabled = isHttp ? disableHttp : (invalidJSON || !values.index.length)`